crates/evm/src/eth/mod.rs

```diff
@@ -119,21 +120,21 @@ where
         self.cfg.chain_id
     }
 
-    fn transact_raw(&mut self, tx: Self::Tx) -> Result<ResultAndState, Self::Error> {
-        if self.inspect {
-            self.inner.set_tx(tx);
-            self.inner.inspect_replay()
-        } else {
-            self.inner.transact(tx)
-        }
+    fn transact_raw(
+        &mut self,
+        tx: Self::Tx,
+    ) -> Result<ResultAndState<ExecutionResult<Self::HaltReason>, EvmState>, Self::Error> {
+        // Use transact_finalize for both inspect and non-inspect paths
+        // In revm v25, inspection is handled internally by the EVM
+        self.inner.transact_finalize(tx)
```

Comment:
I don't think this is correct. It should be something like:
```suggestion
        if self.inspect {
               // inspect_tx_finalize is missing https://github.com/bluealloy/revm/issues/2591
               let output = self.inspect_tx(tx)?;
               let state = self.finalize();
               Ok(ResultAndState::new(output, state))
        } else {
            self.inner.transact_finalize(tx)
        }
```

but lets wait for changes that @klkvr is doing